### PR TITLE
feat(kms): add key_algorithm support

### LINKED ISSUE #1708: [KMS] Add key_algorithm parameter support
I'd like to add `key_algorithm` parameter support to help customers specify Key algorithm

### PR DESCRIPTION
**Pull request by niuzhenguo**
**What this PR does / why we need it**:

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #1708

**Special notes for your reviewer**:

**Release note**:

```release-note

```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST=./huaweicloud/ TESTARGS='-run=TestAccKmsKey_Basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/ -v -run=TestAccKmsKey_Basic -timeout 360m -parallel 4
=== RUN   TestAccKmsKey_Basic
=== PAUSE TestAccKmsKey_Basic
=== CONT  TestAccKmsKey_Basic
--- PASS: TestAccKmsKey_Basic (91.87s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud       91.907s
```

### CODE REVIEW

File: docs/resources/kms_key.md [hunk 1]
```diff
@@ -30,6 +30,9 @@ The following arguments are supported:
 * `key_description` - (Optional, String) The description of the key as viewed in Huawei console. Changing this updates
   the description of key.
 
+* `key_algorithm` - (Optional, String) The algorithm of the key. Valid values are AES_256, SM4, RSA_2048, RSA_3072,
+  RSA_4096, EC_P256, EC_P384, SM2. Changing this creates a new key.
+
```
Comment (Jason-Zhang9309) [inline docs/resources/kms_key.md:35]: missing ForceNew

File: docs/resources/kms_key.md [hunk 2]
```diff
@@ -30,6 +30,9 @@ The following arguments are supported:
 * `key_description` - (Optional, String) The description of the key as viewed in Huawei console. Changing this updates
   the description of key.
 
+* `key_algorithm` - (Optional, String) The algorithm of the key. Valid values are AES_256, SM4, RSA_2048, RSA_3072,
+  RSA_4096, EC_P256, EC_P384, SM2. Changing this creates a new key.
```
Comment (Jason-Zhang9309) [inline docs/resources/kms_key.md:34]: missing ForceNew

Comment (niuzhenguo) [reply docs/resources/kms_key.md:34]: done

File: huaweicloud/data_source_huaweicloud_kms_key_v1.go [hunk 3]
```diff
@@ -173,7 +173,6 @@ func dataSourceKmsKeyV1Read(d *schema.ResourceData, meta interface{}) error {
 	d.Set("scheduled_deletion_date", key.ScheduledDeletionDate)
 	d.Set("key_state", key.KeyState)
 	d.Set("default_key_flag", key.DefaultKeyFlag)
-	d.Set("expiration_time", key.ExpirationTime)
```
Comment (ShiChangkuo) [inline huaweicloud/data_source_huaweicloud_kms_key_v1.go]: It's better to add the `expiration_time` into schema rather than deleting it

Comment (niuzhenguo) [reply huaweicloud/data_source_huaweicloud_kms_key_v1.go]: done

**Pull request merged**